pkg/services/notification/resource_control/notification.go

```diff
@@ -54,44 +58,89 @@ func UpdateNotificationStatus(notificationId, status string) error {
 	return nil
 }
 
-func GetNotification(notificationId string) (*models.Notification, error) {
-	db := globalcfg.GetInstance().GetDB()
-	nf := new(models.Notification)
-	err := db.Where("notification_id = ?", notificationId).First(nf).Error
-	if err != nil {
-		return nil, err
-	}
-	return nf, nil
-}
-
 func DescribeNotifications(ctx context.Context, req *pb.DescribeNotificationsRequest) ([]*models.Notification, uint64, error) {
 	req.NotificationId = stringutil.SimplifyStringList(req.NotificationId)
 	req.ContentType = stringutil.SimplifyStringList(req.ContentType)
 	req.Owner = stringutil.SimplifyStringList(req.Owner)
 	req.Status = stringutil.SimplifyStringList(req.Status)
 
-	limit := dbutil.GetLimit(req.Limit)
-	offset := dbutil.GetOffset(req.Offset)
+	offset := pbutil.GetOffsetFromRequest(req)
+	limit := pbutil.GetLimitFromRequest(req)
 
 	var nfs []*models.Notification
 	var count uint64
 
-	if err := dbutil.GetChain(globalcfg.GetInstance().GetDB().Table(models.TableNotification)).
+	if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableNotification)).
 		AddQueryOrderDir(req, models.NfColCreateTime).
 		BuildFilterConditions(req, models.TableNotification).
 		Offset(offset).
 		Limit(limit).
 		Find(&nfs).Error; err != nil {
-		logger.Errorf(ctx, "Describe Notifications failed: %+v", err)
+		logger.Errorf(ctx, "Describe notifications failed, [%+v]", err)
 		return nil, 0, err
 	}
 
-	if err := dbutil.GetChain(globalcfg.GetInstance().GetDB().Table(models.TableNotification)).
+	if err := nfdb.GetChain(global.GetInstance().GetDB().Table(models.TableNotification)).
 		BuildFilterConditions(req, models.TableNotification).
 		Count(&count).Error; err != nil {
-		logger.Errorf(ctx, "Describe Notifications count failed: %+v", err)
+		logger.Errorf(ctx, "Describe notifications count failed, [%+v]", err)
 		return nil, 0, err
 	}
 
 	return nfs, count, nil
 }
+
+func UpdateNotifications2Pending(ctx context.Context, nfIds []string) ([]string, error) {
```

Comment:
Maybe UpdateNotificationsStatus will be better.
I saw UpdateNotificationsStatus exist.

Reply:
updated.